extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/OpenConnections.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.websockets.next;
+
+import java.util.Collection;
+import java.util.stream.Stream;
+
+import io.smallrye.common.annotation.Experimental;
+
+/**
+ * Provides convenient access to all open connections from clients to {@link WebSocket} endpoints on the server.
+ *
+ * @see WebSocketConnection
+ */
+@Experimental("This API is experimental and may change in the future")
+public interface OpenConnections {
+
+    /**
+     *
+     * @return an immutable collection of all open connections
+     */
+    Collection<WebSocketConnection> get();
+
+    /**
+     *
+     * @param id
+     * @return the open connection or {@code null} if no open connection with the given id exists
+     * @see WebSocketConnection#id()
+     */
+    WebSocketConnection get(String id);
```

Comment:
Is this `id` the `endpointId` or the `WebSocketConnection#id`?

Reply:
It's `WebSocketConnection#id()`, see javadoc. Also it would not make sense to obtain a single connection for an endpoint. But I agree that it would not hurt to rename `id` to `connectionId`. Will do it ;-)